cmd/shell.go

```diff
@@ -1,50 +1,124 @@
-			utils.PrintlnError(err)
-			return
+		if currentContext.ApplicationId != "" && currentContext.ApplicationName != "" &&
+			currentContext.EnvironmentId != "" && currentContext.EnvironmentName != "" &&
+			currentContext.ProjectId != "" && currentContext.ProjectName != "" &&
+			currentContext.OrganizationId != "" && currentContext.OrganizationName != "" {
+			utils.PrintlnInfo("Current context:")
+			if err := utils.PrintlnContext(); err != nil {
+				fmt.Println("Context not yet configured.")
+			}
+			fmt.Println()
+
+			utils.PrintlnInfo("Continue with shell command using this context ?")
+			useContext = utils.Validate("context")
+			fmt.Println()
+		} else {
+			utils.PrintlnInfo("Current context:")
+			if err := utils.PrintlnContext(); err != nil {
+				fmt.Println("Context not yet configured.")
+				fmt.Println("Unable to use current context for `shell` command.")
+				fmt.Println()
+			}
 		}
-		utils.PrintlnInfo("Select environment")
-		env, err := utils.SelectEnvironment(project.ID)
-		if err != nil {
-			utils.PrintlnError(err)
-			return
+
+		var req *pkg.ShellRequest
+		if useContext {
+			req, err = shellRequestFromContext(currentContext)
+		} else {
+			req, err = shellRequestFromSelect()
 		}
-		utils.PrintlnInfo("Select application")
-		app, err := utils.SelectApplication(env.ID)
 		if err != nil {
 			utils.PrintlnError(err)
 			return
 		}
 
-		pkg.ExecShell(&pkg.ShellRequest{
-			ApplicationID:  app.ID,
-			ProjectID:      project.ID,
-			OrganizationID: orga.ID,
-			EnvironmentID:  env.ID,
-			ClusterID:      env.ClusterID,
-		})
+		pkg.ExecShell(req)
 	},
 }
 
+func shellRequestFromSelect() (*pkg.ShellRequest, error) {
+	utils.PrintlnInfo("Select organization")
+	orga, err := utils.SelectOrganization()
+	if err != nil {
+		return nil, err
+	}
+
+	utils.PrintlnInfo("Select project")
+	project, err := utils.SelectProject(orga.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	utils.PrintlnInfo("Select environment")
+	env, err := utils.SelectEnvironment(project.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	utils.PrintlnInfo("Select application")
+	app, err := utils.SelectApplication(env.ID)
+	if err != nil {
+		return nil, err
+	}
+
+	return &pkg.ShellRequest{
+		ApplicationID:  app.ID,
+		ProjectID:      project.ID,
+		OrganizationID: orga.ID,
+		EnvironmentID:  env.ID,
+		ClusterID:      env.ClusterID,
+	}, nil
+}
+
+func shellRequestFromContext(currentContext utils.QoveryContext) (*pkg.ShellRequest, error) {
+	token, err := utils.GetAccessToken()
+	if err != nil {
+		return nil, err
+	}
+
+	auth := context.WithValue(context.Background(), qovery.ContextAccessToken, string(token))
+	client := qovery.NewAPIClient(qovery.NewConfiguration())
+
+	e, res, err := client.EnvironmentMainCallsApi.GetEnvironment(auth, string(currentContext.EnvironmentId)).Execute()
+	if err != nil {
+		return nil, err
+	}
+	if res.StatusCode >= 400 {
+		return nil, errors.New("Received " + res.Status + " response while listing organizations. ")
```

Comment:
Seemed to be `while fetching environment`

Reply:
big copy pasta here, nice catch